recipes/libmysqlclient/all/conanfile.py

```diff
@@ -49,7 +49,10 @@ def _patch_files(self):
         os.rename(sources_cmake, sources_cmake_orig)
         os.rename("CMakeLists.txt", sources_cmake)
         if self.settings.os == "Macos":
-            tools.replace_in_file(os.path.join(self._source_subfolder, "libmysql", "CMakeLists.txt"), "COMMAND $<TARGET_FILE:libmysql_api_test>", "COMMAND DYLD_LIBRARY_PATH=%s $<TARGET_FILE:libmysql_api_test>" % os.path.join(self.build_folder, "library_output_directory"))
+            if self.version <= "8.0.17":
```

Comment:
```suggestion
            if tools.Version(self.version) <= "8.0.17":
```

self.version is a string, need to convert to Version class.